include/envoy/upstream/cluster_manager.h

```diff
@@ -359,6 +360,8 @@ class ClusterManagerFactory {
   virtual Http::ConnectionPool::InstancePtr
   allocateConnPool(Event::Dispatcher& dispatcher, HostConstSharedPtr host,
                    ResourcePriority priority, std::vector<Http::Protocol>& protocol,
+                   const absl::optional<envoy::config::core::v3::AlternateProtocolsCacheOptions>&
```

Comment:
Should this use an OptRef? (I see this is a reference to an optional as opposed to an optional reference, but I suspect they have the same semantics?)

Reply:
I think this (optional<Proto>&) is the type that protobufs return for optional fields. For example, here's the code from cluster_manager_impl.cc:

  absl::optional<envoy::config::core::v3::AlternateProtocolsCacheOptions>
      alternate_protocol_options = host->cluster().alternateProtocolsCacheOptions();

So I think this seems like the right choice?

Reply:
Fine by me, thanks for the explanation.